pkg/maps/lbmap/lbmap.go

```diff
@@ -68,6 +73,30 @@ func (*LBBPFMap) UpsertService(
 
 	slot := 1
 	svcVal := svcKey.NewValue().(ServiceValue)
+
+	if useMaglev && len(backends) != 0 {
+		var backendIDs []uint16
+
+		backendNames := make([]string, 0, len(backends))
+		for name := range backends {
+			backendNames = append(backendNames, name)
+		}
+		m := uint64(option.Config.MaglevTableSize)
+		table := maglev.GetLookupTable(backendNames, m)
+		backendIDs = make([]uint16, m)
+		for i, pos := range table {
+			backendIDs[i] = backends[backendNames[pos]]
+		}
+
+		if err := updateMaglevTable(ipv6, svcID, backendIDs); err != nil {
```

Comment:
For my understanding, this is the only place that the mapping will change? For example if a backend goes away, `UpsertService` will be called?

Reply:
> For example if a backend goes away, UpsertService will be called?

Yes, it will be called with a new set of backends. When a svc is removed, then `DeleteService` will be called which should remove the mappings.